sdk/identity/Azure.Identity/src/CredentialUnavailableException.cs

```diff
@@ -59,6 +59,9 @@ internal static CredentialUnavailableException CreateAggregateException(string m
         /// <param name="info">The <see cref="SerializationInfo"/>.</param>
         /// <param name="context">The <see cref="StreamingContext"/>.</param>
         /// <returns></returns>
+#if NET6_0_OR_GREATER
+        [Obsolete(DiagnosticId = "SYSLIB0051")]
```

Comment:
See [SYSLIB0051: Legacy serialization support APIs are obsolete](https://learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/syslib0051)

Reply:
Do you know why Azure.Identity has these constructors?  Do they need them for some reason?

Reply:
All custom exceptions were advised to have these constructors in the past.  It was part of the guidance for writing custom exceptions before it was removed in the modern platform.